data/build.gradle

```diff
@@ -0,0 +1,33 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+android {
+    compileSdkVersion versionNumbers.compileSdkVersion
+    buildToolsVersion versionNumbers.buildToolsVersion
+    defaultConfig {
+        minSdkVersion versionNumbers.minSdkVersion
+        targetSdkVersion versionNumbers.targetSdkVersion
+        versionCode 1
+        versionName "1.0"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    //Modules
+    implementation project(':domain')
+
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.core:core-ktx:1.1.0'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
```

Comment:
aqui puedes reusar los valores de dependencies.gradle como esta por ejemplo en `app/build.gradle`
`implementation androidXDependencies.values()`

Reply:
Muy bien visto, @DavidCruzUK. Ya he pusheado los cambios :blush:

Reply:
Me acabo de dar cuenta que no era de lo que hice yo... :sweat_smile: 
Pero bueno, para esto también aplica

Reply:
Acabo de hacer otra PR

Reply:
ok, voy a cambiarlo y hago push otra vez :)

Reply:
genial, voy a hecharle un vistazo.

Reply:
Una cosa que se me a olvidado comentar, creo wue como esas dependencias las tenemos en en la app, tendriamos que cambair `implementation` por `api`.
aqui dejos un articulo al respecto: https://medium.com/mindorks/implementation-vs-api-in-gradle-3-0-494c817a6fa
que os parece?

Reply:
por lo que he leido en el articulo(lo he leido rapido y por encima) creo q la mejor opcion es seguir utilizando implementation, primero porque de esa manera nos aseguramos de q un modulo pueda leer solo los modulos que tiene implementados y no los modulos de estos tambien, de esta forma el ui module por ejemplo, no podria leer nada de la capa de domain o data solo la de presentation,  reforzando asi el clear-architecture. Segundo porque con api no solo recompilaria  el modulo al que apunta sino todos los  modulos de este tambien

Reply:
es verdad  ahora lo he puesto asi en el build gradle de manera temporal
implementation project(':presentation')
    implementation project(':domain')
    implementation project(':data'):
pero puede ser que no necesitemos alguna de estas tres en un futuro, posiblemente data, as refuerzas q esta capa no puede ser accedida de una activity por ejemplo

Reply:
Lo que comenta @gabrielpozo de los tiempos de compilación, también lo he leído. Y por cierto, es un fastidio que cada vez que añades una `dynamic feature` con el "wizard" de Android Studio, te cambia los `implemenation` por `api`.

Reply:
Os dejo una respuesta de stackoverflow que he leído que puede servir para aclarar el tema de _implementation_ vs _api_. Es una respuesta bastante breve, pero viene a decir lo mismo que el articulo de medium que ha publicado David.

https://stackoverflow.com/a/44419574/4529022

Yo lo veo como comenta @gabrielpozo, creo que lo ideal sería hacerlo a través de implementation para separarnos lo máximo posible de otras capas.

PD: No conocía exactamente la diferencia que había entre utilizar _implementation_ o _api_ y me parece brutal https://giphy.com/gifs/5VKbvrjxpVJCM

Reply:
¡¡Hostia, qué cracks!!

Acabo de entender el por qué tenía que meter cada uno de los módulos que usaba en una dynamic feature (aunque la app tuviera esos módulos ya metidos). No sabía el por qué.

Muchísimas gracias. Ahora ya lo entiendo... :-)
¡¡¡Qué alegría!!! :-D